src/services/ClaimedIdRequestProcessor.ts

```diff
@@ -75,19 +76,22 @@ export class ClaimedIdRequestProcessor {
 
 			this.metrics.addMetric("Found session", MetricUnits.Count, 1);
 
-			if (session.authSessionState !== AuthSessionState.CIC_SESSION_CREATED) {
-				this.logger.error(`Session is in the wrong state: ${session.authSessionState}, expected state should be ${AuthSessionState.CIC_SESSION_CREATED}`, { 
-					messageCode: MessageCodes.INCORRECT_SESSION_STATE,
-				});
-				return new Response(HttpCodesEnum.UNAUTHORIZED, `Session is in the wrong state: ${session.authSessionState}`);
+			switch (session.authSessionState) {
+			  case AuthSessionState.CIC_SESSION_CREATED:
+					await this.cicService.saveCICData(sessionId, cicSession, session.expiryDate);
+					return new Response(HttpCodesEnum.OK, "");
+			  case AuthSessionState.CIC_DATA_RECEIVED:
+			  case AuthSessionState.CIC_AUTH_CODE_ISSUED:
+			  case AuthSessionState.CIC_ACCESS_TOKEN_ISSUED:
+					this.logger.info("Duplicate request, returning status 200, sessionId: ", sessionId);
+					return new Response(HttpCodesEnum.OK, "");
```

Comment:
Could we add a message to these responses - maybe something like "Request already processed" and "Request successfully processed"?

Reply:
Done